Packs/ThreatGrid/Integrations/ThreatGrid/ThreatGrid_test.py

```diff
@@ -1,174 +1,615 @@
+    """
+
+    from ThreatGrid import list_associated_samples_command
+
+    mock_response = load_mock_response('associated_samples_list.json')
+
+    requests_mock.get(url=url, json=mock_response)
+
+    result = list_associated_samples_command(mock_client, args)
+
+    assert result.outputs_prefix == outputs_prefix
+    assert result.outputs['samples'][0]['sha256'] == 'sha256'  # type: ignore[assignment]
+
+
+@pytest.mark.parametrize('url, args,outputs_prefix', [
+    (f'/{API_VERSION2_URL}/samples/sample_id/analysis/annotations', {
+        'sample_id': 'sample_id',
+        'command_name': 'threat-grid-analysis-annotations-get'
+    }, 'SampleAnnotations'),
+    (f'/{API_VERSION2_URL}/samples/sample_id/analysis/artifacts/artifact', {
+        'sample_id': 'sample_id',
+        'artifact_id': 'artifact',
+        'command_name': 'threat-grid-analysis-artifacts-get'
+    }, 'ArtifactAnalysis'),
+    (f'/{API_VERSION2_URL}/samples/sample_id/analysis/iocs', {
+        'sample_id': 'sample_id',
+        'ioc': 'ioc',
+        'command_name': 'threat-grid-analysis-iocs-get'
+    }, 'IOCAnalysis'),
+    (f'/{API_VERSION2_URL}/samples/sample_id/analysis/metadata', {
+        'sample_id': 'sample_id',
+        'command_name': 'threat-grid-analysis-metadata-get'
+    }, 'AnalysisMetadata'),
+    (f'/{API_VERSION2_URL}/samples/sample_id/analysis/network_streams/network_stream', {
+        'sample_id': 'sample_id',
+        'network_stream_id': 'network_stream',
+        'command_name': 'threat-grid-analysis-network-streams-get'
+    }, 'NetworkAnalysis'),
+    (f'/{API_VERSION2_URL}/samples/sample_id/analysis/processes/process_id', {
+        'sample_id': 'sample_id',
+        'process_id': 'process_id',
+        'command_name': 'threat-grid-analysis-processes-get'
+    }, 'ProcessAnalysis'),
+])
+def test_analysis_sample_command(requests_mock, mock_client, url, args, outputs_prefix):
+    """
+    Scenario: Get data about a specific IOC / processes / artifact / network-stream
+        from the relevant section of the sample's analysis.json.
+    Given:
+     - User has provided valid credentials.
+    When:
+     - threat-grid-analysis-annotations-get called.
+     - threat-grid-analysis-artifacts-get called.
+     - threat-grid-analysis-iocs-get called.
+     - threat-grid-analysis-metadata-get called.
+     - threat-grid-analysis-network-streams-get called.
+     - threat-grid-analysis-processes-get called.
+    Then:
+     - Ensure number of items is correct.
+     - Ensure outputs prefix is correct.
+     - Ensure a sample value from the API matches what is generated in the context.
+    """
+
+    from ThreatGrid import analysis_sample_command
+
+    mock_response = load_mock_response('sample_analysis.json')
+
+    requests_mock.get(url=url, json=mock_response)
+
+    result = analysis_sample_command(mock_client, args)
+
+    assert result.outputs_prefix == f'ThreatGrid.{outputs_prefix}'
```

Comment:
this only checks the outputs prefix, and doesn't check all the things the docstrings declar.
Please see `ipinfo_v2` for a unit test example.